src/plugins/bodhi.c

```diff
@@ -538,41 +535,40 @@ int main(int argc, char **argv)
             if (!problem_data)
                 libreport_xfunc_die(); /* create_problem_data_for_reporting already emitted error msg */
 
-            char *product, *version;
-            map_string_t *osinfo = libreport_new_map_string();
+            g_autofree char *product = NULL;
+            g_autofree char *version = NULL;
+            map_string_t *osinfo = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, g_free);
```

Comment:
You could probably opt for automatic cleanup (`map_string_t` is just a typedef for `GHashTable`).